examples/ex0.cpp

```diff
@@ -0,0 +1,97 @@
+//                                MFEM Example 0
+//
+// Compile with: make ex0
+//
+// Sample runs:  ex0
+//               ex0 -m ../data/fichera.mesh
+//               ex0 -m ../data/square-disc.mesh -o 2
+//
+// Description: This example code demonstrates the most basic usage of MFEM
+//              to define a simple finite element discretization of the Laplace
+//              problem -Delta u = 1 with homogeneous Dirichlet boundary
+//              conditions. The mesh file and finite element polynomial degree
+//              are given by command line options.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command line options
+   const char *mesh_file = "../data/star.mesh";
+   int order = 1;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh", "Mesh file to use.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree)");
```

Comment:
Would it make sense to have some `DefaultOption` which are the common options used by most examples?

Reply:
Personally I don't think so. I think it's better to show the options explicitly rather than have it hidden in `optparser.cpp`.